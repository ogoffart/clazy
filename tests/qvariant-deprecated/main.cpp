#include <QtCore/QVariant>

int foo(const QVariant &v) {
    switch (v.type()) {
        case QVariant::Map: return 0;
        case QVariant::List: return 1;
        case QVariant::Int: return 2;
        default: break;
    }

    if (v.type() == QVariant::Line) return 3;
    return 4;
}
