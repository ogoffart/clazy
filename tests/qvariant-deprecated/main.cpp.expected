qvariant-deprecated/main.cpp:4:13: warning: Use QVariant::userType() instead of QVariant::type() [-Wclazy-qvariant-deprecated]
qvariant-deprecated/main.cpp:5:14: warning: QVariant::Map is deprecated, use QMetaType::QVariantMap [-Wclazy-qvariant-deprecated]
qvariant-deprecated/main.cpp:6:14: warning: QVariant::List is deprecated, use QMetaType::QVariantList [-Wclazy-qvariant-deprecated]
qvariant-deprecated/main.cpp:7:14: warning: QVariant::Int is deprecated, use QMetaType::Int [-Wclazy-qvariant-deprecated]
qvariant-deprecated/main.cpp:11:9: warning: Use QVariant::userType() instead of QVariant::type() [-Wclazy-qvariant-deprecated]
qvariant-deprecated/main.cpp:11:21: warning: QVariant::Line is deprecated, use QMetaType::QLine [-Wclazy-qvariant-deprecated]
